src/app/services/ws.service.ts

```diff
@@ -0,0 +1,33 @@
+import { SocketIOController, WebsocketContext, wsController } from '@foal/socket.io';
+import { ChatController } from '../controllers/api/chat.controller'
+import { Config } from '@foal/core';
+
+const prefix = Config.get('api_prefix', 'string', '');
+
+export class WebsocketService extends SocketIOController {
+    constructor() {
+        super();
+        this.options = {
+            cors: {
+                origin: '*', 
+            },
+            path: `${prefix}/api/hci-socket`
```

Comment:
This isn't really a big deal, but "hci-socket" is maybe a bit of a weird name, since we already know this is the HCI API, and "socket" is maybe a bit generic (eg, we already have n "hci socket" that refers to the unix socket the application is listening on. You could also make an argument that the websocket endpoint should not be under `api` since it is not part of the REST API. If it was me, I might name it `${prefix}/realtime`, though you may  not be in a position to change it now.

Reply:
I can change the path no prob ill just coordinate with the person who revamp the frontend to change the path as well. Opinions on  `${prefix}/realtime-socket`? Id like to keep the `-socket` onto the path

Reply:
Personally I think specifying `socket` is extraneous (I think the semantics take care of it, and in general I try to stick wit names for things describing purpose rather than type), but `${prefix}/realtime-socket` is fine if you think that's better.